src/common/thread_measurement.cpp

```diff
@@ -53,6 +80,6 @@ void StartMeasurementTask(void const *argument) {
             next_sg_cycle = now + next_delay;
         }
 
-        osDelay(checkTimestampsAscendingOrder(next_fs_cycle, next_sg_cycle) ? next_fs_cycle - now : next_sg_cycle - now);
+        osDelay(std::max({ next_fs_cycle, next_sg_cycle, next_fs_autoload_cycle }) - now);
```

Comment:
`checkTimestampsAscendingOrder` properly handled the case of uint32_t overflowing; this ignores the problem :)

Btw – shouldn't it be std::min? You want to wake up with the closest event, not on the one being the furthest in the future.